var/spack/repos/builtin/packages/libmesh/package.py

```diff
@@ -32,23 +32,31 @@ class Libmesh(Package):
 
     homepage = "http://libmesh.github.io/"
     url      = "https://github.com/libMesh/libmesh/releases/download/v1.0.0/libmesh-1.0.0.tar.bz2"
+    git      = "https://github.com/libMesh/libmesh.git"
 
+    version('1.3.0', sha256='a8cc2cd44f42b960989dba10fa438b04af5798c46db0b4ec3ed29591b8359786')
+    version('1.2.1', sha256='11c22c7d96874a17de6b8c74caa45d6745d40bf3610e88b2bd28fd3381f5ba70')
     version('1.0.0', 'cb464fc63ea0b71b1e69fa3f5d4f93a4')
 
     variant('mpi', default=True, description='Enables MPI parallelism')
+    variant('slepc', default=False, description='SLEPc eigensolver')
 
     depends_on('mpi', when='+mpi')
 
     # Parallel version of libmesh needs parallel solvers
     depends_on('petsc+mpi', when='+mpi')
 
+    # SLEPc version needs slepc
+    depends_on('slepc', when='+slepc')
+
     def install(self, spec, prefix):
         config_args = ["--prefix=%s" % prefix]
 
         if '+mpi' in spec:
             config_args.append('CC=%s' % spec['mpi'].mpicc)
             config_args.append('CXX=%s' % spec['mpi'].mpicxx)
             config_args.append('PETSC_DIR=%s' % spec['petsc'].prefix)
+            config_args.append('SLEPC_DIR=%s' % spec['slepc'].prefix)
```

Comment:
you will need to guard this by an if statement, in case of `~slepc` this should end up being an KeyError

Reply:
and also if this is `+mpi` only you should add a `conflicts('~mpi', when='+slepc'`)

Reply:
Thanks for the feedback. `+slepc` indeed requires `+mpi`. SLEPc has both PETSc/MPI as required dependencies, but we still need to link against both in libMesh build so I think what I have covers it all properly now.